### PR TITLE
feat: dashboard post history

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #

### CODE REVIEW

File: packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+interface PlaceholderProps {
+    network: string
+}
+
+export const Placeholder = memo<PlaceholderProps>(({ network }) => {
+    const t = useDashboardI18N()
+    const url = urlcat('https://www.:network', { network: network })
+
+    const handleClick = () => Services.Settings.openSNSAndActivatePlugin(url, 'none')
```
Comment (nuanyang233) [inline packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx:16]: useCallback

Comment (nuanyang233) [reply packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx:16]: I suggest `useAsyncFn` for loading state

File: packages/dashboard/src/pages/Personas/components/PostHistory/PostHistoryRow.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { Stack, Typography } from '@material-ui/core'
+import { memo, ReactNode, useMemo } from 'react'
+import { FileMessageIcon, ITOIcon, MessageIcon, RedPacketIcon, PollIcon } from '@masknet/icons'
+import { getMaskColor } from '@masknet/theme'
+import { Services } from '../../../../API'
+
+const SUPPORT_PLUGIN: Record<string, ReactNode> = {
```
Comment (nuanyang233) [inline packages/dashboard/src/pages/Personas/components/PostHistory/PostHistoryRow.tsx:7]: Why does the key not use Enum? 🤔

File: packages/maskbook/src/extension/background-script/CryptoServices/encryptTo.ts [hunk 3]
```diff
@@ -121,11 +121,11 @@ function getSummary(content: TypedMessageText) {
         // it seems like using "en" can also split the word correctly.
         const seg = new Intl.Segmenter('en')
         for (const word of seg.segment(content.content)) {
-            if (result.length >= 20) break
+            if (result.length >= 30) break
```
Comment (nuanyang233) [inline packages/maskbook/src/extension/background-script/CryptoServices/encryptTo.ts:124]: The 30 looks like a magic number 👀, Why do you need to set it to 30

File: packages/dashboard/package.json [hunk 4]
```diff
@@ -40,6 +40,7 @@
     "react-router-dom": "^6.0.0-beta.0",
     "react-use": "^17.3.1",
     "unstated-next": "^1.1.0",
+    "urlcat": "^2.0.4",
```
Comment (nuanyang233) [inline packages/dashboard/package.json:43]: maybe we need to move it to the root package.json

File: packages/dashboard/src/pages/Personas/components/PostHistory/PostHistoryRow.tsx (hunk 2 above)
Comment (NikoManiac) [reply packages/dashboard/src/pages/Personas/components/PostHistory/PostHistoryRow.tsx:7]: I want use it like this: ` SUPPORT_PLUGIN[plugin] ?? <MessageIcon sx={{ width: 48, height: 48 }} />`

File: packages/maskbook/src/extension/background-script/CryptoServices/encryptTo.ts (hunk 3 above)
Comment (NikoManiac) [reply packages/maskbook/src/extension/background-script/CryptoServices/encryptTo.ts:124]: The 20 chars is too short to show the twitter content summary.

File: packages/dashboard/package.json (hunk 4 above)
Comment (guanbinrui) [reply packages/dashboard/package.json:43]: vote +1

Comment (NikoManiac) [reply packages/dashboard/package.json:43]: Only two packages are needed the `urlcat` at this time. And the root package.json is very minimalist.

File: packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    return (
+        <Stack height="100%" alignItems="center" justifyContent="center">
+            <Box textAlign="center">
+                <EmptyIcon sx={{ fontSize: 100 }} />
+                <Typography variant="body2" mb={3}>
+                    {t.personas_post_is_empty()}
+                </Typography>
+                <Button onClick={handleClick}>{t.personas_post_create()}</Button>
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx:25]: Use `ActionButtonPromise`.

File: packages/dashboard/package.json (hunk 4 above)
Comment (Jack-Works) [reply packages/dashboard/package.json:43]: Don't move it to the top. It's the correct way to use pnpm.

File: packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx (hunk 5 above)
Comment (Jack-Works) [reply packages/dashboard/src/pages/Personas/components/PostHistory/Placeholder.tsx:25]: I don't encourage using `ActionButtonPromise` and it's not accessible from the dashboard package too. I'll provide a new API if needed.

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**